Documents/Command/ConfigureAgentLogsUploadToCloudwatch/ConfigureAgentLogsUploadToCloudwatch.json

```diff
@@ -1,257 +1,256 @@
-            "                xmlstarlet ed --inplace -s /seelog/outputs -t elem -n custom -v '' $TEMPFILE",
-            "                xmlstarlet ed --inplace -a /seelog/outputs/custom -t attr -n name -v 'cloudwatch_receiver' $TEMPFILE",
-            "                xmlstarlet ed --inplace -a /seelog/outputs/custom -t attr -n formatid -v $FORMATID $TEMPFILE",
-            "                if [ ! -z $LOGGROUP ]",
-            "                then",
-            "                        xmlstarlet ed --inplace -a /seelog/outputs/custom -t attr -n data-log-group -v $LOGGROUP $TEMPFILE",
-            "                fi",
-            "        fi",
-            "else",
-            "        echo 'Disable CloudWatchLogs Upload'",
-            "        xmlstarlet ed --inplace -d /seelog/outputs/custom $TEMPFILE",
-            "fi",
-            "",
-            "echo 'Applying Changes to Config File'",
-            "mv $TEMPFILE $FILEPATH"
-          ]
-        }
+      "name":"UploadSSMAgentLogsInLinux",
+      "inputs": {
+        "runCommand":[
+          "PYTHON_CMD=''",
+          "which python3 2>/dev/null",
+          "HAS_PYTHON3=$?",
+          "which python 2>/dev/null",
+          "HAS_PYTHON2=$?",
+          "which apt 2>/dev/null",
+          "HAS_APT=$?",
+          "which yum 2>/dev/null",
+          "HAS_YUM=$?",
+          "which zypper 2> /dev/null",
+          "HAS_ZYPP=$?",
+          "",
+          "",
+          "",
+          "check_install_code() {",
+          "    if [ $1 -ne 0 ]",
+          "    then",
+          "        echo \"WARNING: Could not install the $2, this may cause the patching operation to fail.\" >&2",
+          "    fi",
+          "}",
+          "",
+          "if [ $HAS_APT -eq 0 -a $HAS_PYTHON3 -eq 0 ]",
+          "then",
+          "    PYTHON_CMD='python3'",
+          "    apt-get install python3-apt -y",
+          "    check_install_code $? \"python3-apt\"",
+          "    apt-get install python3-six -y",
+          "    check_install_code $? \"python3-six\"",
+          "",
+          "elif [ $HAS_YUM -eq 0 -a $HAS_PYTHON2 -eq 0 ]",
+          "then",
+          "    PYTHON_CMD='python'",
+          "",
+          "elif [ $HAS_ZYPP -eq 0 -a $HAS_PYTHON2 ]",
+          "then",
+          "    PYTHON_CMD='python'",
+          "    zypper install -y python-zypp",
+          "    check_install_code $? \"python-zypp\"",
+          "else",
+          "    echo \"An unsupported package manager and python version combination was found. Yum requires Python2 and Apt requires Python 3.\"",
+          "    echo \"Python3=$HAS_PYTHON3, Python2=$HAS_PYTHON, Yum=$HAS_YUM, Apt=$HAS_APT.\"",
+          "    echo \"Exiting...\"",
+          "    exit 1",
+          "fi",
+          "echo Succeeded to find/install python",
+          
+          "export LOGLEVEL='{{ LogLevel }}'",
+          "export ENABLE={{ EnableCloudWatchUpload }}",
```

Comment:
As explained for the Windows-equivalent script, avoid injecting document variables into your scope without quotes around them and appropriate validation.